### PR TITLE
Don't pull the probe image if we already have it.

### PR DESCRIPTION
**Pull request by negz**
Reviewers: @mattnworb 
Peanuts: @spotify/helios-team 

Its freshness is not a concern.

### CODE REVIEW

**Comment by davidxia** [comment]
Just a general question from me: doesn't `docker pull` first compare the sha256 hashes of the local and remote layers an image is made of and only pulls layer content from a remote registry if any of the hashes are different?

**Comment by gimaker** [comment]
LGTM but any chance we can add tests for this?

**Comment by mattnworb** [comment]
@davidxia I think so. A problem arises though if the network connection to Docker Hub has an issue; for some test users it ended up failing their build and tests, even though they had the alpine image already.

@negz nice change. Agree with Staffan on the test but otherwise :+1:

**Comment by davidxia** [comment]
:+1:

**Comment by mattnworb** [comment]
:+1: for the unit tests

File: helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java [hunk 1]
```diff
@@ -245,6 +247,17 @@ private String checkDockerAndGetGateway() throws HeliosDeploymentException {
     return gateway;
   }
 
+  private void pullIfAbsent(final String image) throws DockerException, InterruptedException {
+    try {
+      dockerClient.inspectImage(image);
+      log.info("helios-solo image {} is present. Not pulling it.", image);
```
Comment (davidxia) [inline helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java]: derp, how did we miss this? The log messages and `pull(PROBE_IMAGE)` are wrong. PR here to fix: https://github.com/spotify/helios/pull/918

**Pull request merged**